config/v1/types_infrastructure.go

```diff
@@ -1010,6 +1010,15 @@ type VSpherePlatformTopology struct {
 	// +kubebuilder:validation:Pattern=`^/.*?/vm/.*?`
 	// +optional
 	Folder string `json:"folder,omitempty"`
+
+	// template is the inventory path of the virtual machine or template
+	// that will be used for cloning.
+	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^/.*?/vm/.*?`
```

Comment:
These validations should be explained in prose within the godoc.

Do we run the same regex validation against the template field on the providerSpec?

Reply:
yes, we will perform that validation in the cpmso